openmc/material.py

```diff
@@ -0,0 +1,1109 @@
+                warnings.warn(msg)
+        else:
+            if density is None:
+                msg = 'Unable to set the density for Material ID="{}" ' \
+                      'because a density value must be given when not using ' \
+                      '"sum" unit'.format(self.id)
+                raise ValueError(msg)
+
+            cv.check_type('the density for Material ID="{}"'.format(self.id),
+                          density, Real)
+            self._density = density
+
+    @distrib_otf_file.setter
+    def distrib_otf_file(self, filename):
+        # TODO: remove this when distributed materials are merged
+        warnings.warn('This feature is not yet implemented in a release '
+                      'version of openmc')
+
+        if not isinstance(filename, str) and filename is not None:
+            msg = 'Unable to add OTF material file to Material ID="{}" with a ' \
+                  'non-string name "{}"'.format(self._id, filename)
+            raise ValueError(msg)
+
+        self._distrib_otf_file = filename
+
+    @convert_to_distrib_comps.setter
+    def convert_to_distrib_comps(self):
+        # TODO: remove this when distributed materials are merged
+        warnings.warn('This feature is not yet implemented in a release '
+                      'version of openmc')
+
+        self._convert_to_distrib_comps = True
+
+    def add_nuclide(self, nuclide, percent, percent_type='ao'):
+        """Add a nuclide to the material
+
+        Parameters
+        ----------
+        nuclide : str
+            Nuclide to add, e.g., 'Mo95'
+        percent : float
+            Atom or weight percent
+        percent_type : {'ao', 'wo'}
+            'ao' for atom percent and 'wo' for weight percent
+
+        """
+        cv.check_type('nuclide', nuclide, str)
+        cv.check_type('percent', percent, Real)
+        cv.check_value('percent type', percent_type, {'ao', 'wo'})
+
+        if self._macroscopic is not None:
+            msg = 'Unable to add a Nuclide to Material ID="{}" as a ' \
+                  'macroscopic data-set has already been added'.format(self._id)
+            raise ValueError(msg)
+
+        # If nuclide name doesn't look valid, give a warning
+        try:
+            Z, _, _ = openmc.data.zam(nuclide)
+        except ValueError as e:
+            warnings.warn(str(e))
+        else:
+            # For actinides, have the material be depletable by default
+            if Z >= 89:
+                self.depletable = True
+
+        self._nuclides.append((nuclide, percent, percent_type))
+
+    def remove_nuclide(self, nuclide):
+        """Remove a nuclide from the material
+
+        Parameters
+        ----------
+        nuclide : str
+            Nuclide to remove
+
+        """
+        cv.check_type('nuclide', nuclide, str)
+
+        # If the Material contains the Nuclide, delete it
+        for nuc in self._nuclides:
+            if nuclide == nuc[0]:
```

Comment:
had to dig a bit to figure out why you're using `nuc[0]` here - just nudging a bit stronger on the namedtuple idea :smiley_cat: